test/DirectEditingSpec.js

```diff
@@ -340,6 +342,117 @@ describe('diagram-js-direct-editing', function() {
         });
       }));
 
+      it('should update with changed text', inject(function(canvas, directEditing, directEditingProvider) {
+        // given
+        sinon.spy(directEditingProvider, 'update');
+
+        var shapeWithLabel = {
+          id: 's1',
+          x: 20, y: 10, width: 60, height: 50,
+          label: 'FOO',
+          labelBounds: { x: 100, y: 200, width: 300, height: 20 }
+        };
+
+        canvas.addShape(shapeWithLabel);
+
+        // when
+        directEditing.activate(shapeWithLabel);
+
+        var textbox = directEditing._textbox;
+
+        textbox.content.innerText = 'BAR';
+
+        directEditing.complete();
+
+        // then
+        expect(directEditingProvider.update.callCount).to.equal(1);
+
+        directEditingProvider.update.restore();
+      }));
+
+
+      it('should update with changed bounds height', inject(function(canvas, directEditing, directEditingProvider) {
+        // given
+        sinon.spy(directEditingProvider, 'update');
+
+        var shapeWithLabel = {
+          id: 's1',
+          x: 20, y: 10, width: 60, height: 50,
+          label: 'FOO',
+          labelBounds: { x: 100, y: 200, width: 300, height: 20 }
+        };
+
+        canvas.addShape(shapeWithLabel);
+
+        // when
+        directEditing.activate(shapeWithLabel);
+
+        var textbox = directEditing._textbox;
+
+        textbox.parent.style.height = '21px';
+
+        directEditing.complete();
+
+        // then
+        expect(directEditingProvider.update.callCount).to.equal(1);
+
+        directEditingProvider.update.restore();
+      }));
+
+      it('should update with changed bounds width', inject(function(canvas, directEditing, directEditingProvider) {
+        // given
+        sinon.spy(directEditingProvider, 'update');
+
+        var shapeWithLabel = {
+          id: 's1',
+          x: 20, y: 10, width: 60, height: 50,
+          label: 'FOO',
+          labelBounds: { x: 100, y: 200, width: 300, height: 20 }
+        };
+
+        canvas.addShape(shapeWithLabel);
+
+        // when
+        directEditing.activate(shapeWithLabel);
+
+        var textbox = directEditing._textbox;
+
+        textbox.parent.style.width = '301px';
+
+        directEditing.complete();
+
+        // then
+        expect(directEditingProvider.update.callCount).to.equal(1);
+
+        directEditingProvider.update.restore();
+      }));
+
+
+      it('should NOT update with unchanged text, bounds height or bounds width', inject(function(canvas, directEditing, directEditingProvider) {
```

Comment:
This line is way beyond 80 chars, can you put the function on another line:

```
it('foo bar', inject(
  function(a, b, c) {
    // test code
  }
));
```

Reply:
Shall we consider adding this to our bpmn-io/recommended plugin via the `max-len` [rule](https://eslint.org/docs/rules/max-len)? There's a handful of other lines in directEditingSpec.js, before this PR, which exceed 80 chars.

The rule understandably doesn't support auto-fixing though so that might take quite some time to fix across different projects.

Reply:
For now, let's keep the rules as they are now. We use https://github.com/bpmn-io/eslint-plugin-bpmn-io for our linters and you could propose the rule there. It is outside of the PR's scope, though.